landoapi/api/treestatus.py

```diff
@@ -0,0 +1,609 @@
+            type="https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400",
+        )
+
+    new_tree = Tree(
+        tree=tree,
+        status=body["status"],
+        reason=body["reason"],
+        message_of_the_day=body["message_of_the_day"],
+    )
+    try:
+        session.add(new_tree)
+        session.commit()
+    except (sa.exc.IntegrityError, sa.exc.ProgrammingError):
+        raise ProblemException(
+            400,
+            f"Tree {tree} already exists.",
+            "Tree already exists.",
+            type="https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400",
+        )
+
+    return new_tree.to_dict(), 200
+
+
+@auth.require_auth0(
+    groups=(auth.TREESTATUS_ADMIN), scopes=("lando", "profile", "email"), userinfo=True
+)
+def delete_tree(tree: str) -> tuple[str, int]:
+    """Handler for `DELETE /trees/{tree}`."""
+    remove_tree_by_name(tree)
+    return tree, 200
+
+
+@auth.require_auth0(
+    groups=(auth.TREESTATUS_USERS), scopes=("lando", "profile", "email"), userinfo=True
+)
+def update_log(id: int, body: dict):
+    """Handler for `PATCH /log/{id}`."""
+    session = db.session
+
+    tags = body.get("tags")
+    reason = body.get("reason")
+
+    if tags is None and reason is None:
+        return
+
+    # Update the log table.
+    update_tree_log(id, tags, reason)
+
+    # Iterate over all stack.
+    for status_change in StatusChange.query.all():
+        for tree in status_change.trees:
+            last_state = load_last_state(tree.last_state)
+
+            if last_state["current_log_id"] != id:
+                continue
+
+            if reason:
+                last_state["current_reason"] = reason
+            if tags:
+                last_state["current_tags"] = tags
+
+            tree.last_state = json.dumps(last_state)
+
+    session.commit()
+
+    return None, 200
+
+
+def get_logs_for_tree(tree_name: str, limit_logs: bool = True) -> list[dict]:
+    """Return a list of Log entries as dicts.
+
+    If `limit_logs` is `True`, limit the number of returned logs to the log limit.
+    """
+    # Verify the tree exists first.
+    tree = Tree.query.filter_by(tree=tree_name).one_or_none()
+    if not tree:
+        raise ProblemException(
+            404,
+            f"No tree {tree_name} found.",
+            "Could not find the requested tree.",
+            type="https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/404",
+        )
+
+    query = Log.query.filter_by(tree=tree_name).order_by(Log.created_at.desc())
+    if limit_logs:
+        query = query.limit(TREE_SUMMARY_LOG_LIMIT)
+
+    return [log.to_dict() for log in query]
+
+
+@result_object_wrap
+def get_logs_all(tree: str) -> list[dict]:
+    """Handler for `GET /trees/{tree}/logs_all`."""
+    return get_logs_for_tree(tree, limit_logs=False)
+
+
+@result_object_wrap
+def get_logs(tree: str) -> list[dict]:
+    """Handler for `GET /trees/{tree}/logs`."""
+    return get_logs_for_tree(tree, limit_logs=True)
```

Comment:
I assume the reason we are using a different URL here is for compatibility? It would be a bit nicer to have a single endpoint (instead of this one and `get_logs_all`, and just pass a limit parameter instead.

Reply:
Yes, it's for compatibility, since this is a publicly accessible endpoint. Once we migrate over I'm going to re-audit usage of the endpoints and hopefully we can consolidate into a single endpoint.